server/src/mds3.init.js

```diff
@@ -1762,6 +1770,49 @@ async function validate_query_singleSampleGenomeQuantification(ds, genome) {
 	}
 }
 
+async function validate_query_NIdata(ds, genome) {
+	const q = ds.queries.NIdata
+	if (!q || !serverconfig.features?.showBrainImaging) return
+	for (const key in q) {
+		if (q[key].referenceFile && q[key].samples) {
+			q[key].get = async (sampleName, l, f, t) => {
+				const refFile = path.join(serverconfig.tpmasterdir, q[key].referenceFile)
+				const sampleFile = path.join(serverconfig.tpmasterdir, q[key].samples, sampleName)
+
+				try {
+					await fs.promises.stat(sampleFile)
+				} catch (e) {
+					if (e.code == 'EACCES') throw 'cannot read file, permission denied'
+					if (e.code == 'ENOENT') throw 'no data for this sample'
+					throw 'failed to load data'
+				}
+
+				return new Promise((resolve, reject) => {
+					const ps = spawn('python3', ['proteinpaint/python/src/plotBrainImaging.py', refFile, sampleFile, l, f, t])
+					let imgData = []
+					ps.stdout.on('data', data => {
+						imgData.push(data)
+					})
+					ps.stderr.on('data', data => console.error(`stderr: ${data}`))
```

Comment:
The promise should be rejected here too:

ps.stderr.on('data', data =>  {
console.error(`stderr: ${data}`
reject(new Error(`Python script filed: ${data}`))
}))